src/preprocessing/preprocessing-modules/HTMLTag.ts

```diff
@@ -0,0 +1,37 @@
+import { Config } from '../../config';
+import { parseHTML, serializeHTML } from '../util';
+
+const HTML_ATTRIBUTES = [
+  'allow-xhr-interception',
+  'allow-viewer-render-template',
+  'report-errors-to-viewer',
+];
+
+/**
+ * Adds attributes to the <html> tag which affect which features are allowed by
+ * the AMP viewer.
+ *
+ * @param {string} amp AMP code to modify <html> tag of
+ * @param {!Config} config Global config
+ * @return {string} Modified AMP code
+ */
+function process(amp: string, config: Config): string {
+  const doc = parseHTML(amp);
+
+  const tags = doc.getElementsByTagName('html');
+  if (tags.length !== 1) {
+    throw new Error('Failed to find <html> tag inside AMP document');
+  }
+  const html = tags[0];
+
+  for (const attr of HTML_ATTRIBUTES) {
```

Comment:
how would one opt out of applying some of these attributes? I think that logic needs to be in place.

Reply:
I thought about adding it to the config, but I'm not sure if it should be done on that level?

In theory, these could still be decided on the viewer capability level, adding them is just a prerequisite if I understand correctly, but won't automatically enable those features.

Reply:
That is correct, it is a prereq and a capability that needs to be set by the viewer so yes, this is fine then.